libs/auth/react/src/lib/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,260 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  ReactNode,
+  useState,
+} from 'react'
+import type { User } from 'oidc-client-ts'
+import { Location } from 'react-router-dom'
+import { getAuthSettings, getUserManager } from '../userManager'
+import { ActionType, initialState, reducer } from './Auth.state'
+import { AuthSettings } from '../AuthSettings'
+import { AuthContext } from './AuthContext'
+import { AuthErrorScreen } from './AuthErrorScreen'
+import { AuthLoadingScreen } from './AuthLoadingScreen'
+import { CheckIdpSession } from './CheckIdpSession'
+import { isDefined } from '@island.is/shared/utils'
+
+interface AuthProviderProps {
+  /**
+   * If true, Authenticator automatically starts login flow and does not render children until user is fully logged in.
+   * If false, children are responsible for rendering a login button and loading indicator.
+   * Default: true
+   */
+  autoLogin?: boolean
+  /**
+   * The base path of the application.
+   */
+  basePath: string
+  children: ReactNode
+}
+
+const getReturnUrl = (returnUrl: string, { redirectPath }: AuthSettings) => {
+  if (redirectPath && returnUrl.startsWith(redirectPath)) {
+    return '/'
+  }
+
+  return returnUrl
+}
+
+export const AuthProvider = ({
+  children,
+  autoLogin = true,
+  basePath,
+}: AuthProviderProps) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const [hasError, setHasError] = useState(false)
+  const userManager = getUserManager()
+  const authSettings = getAuthSettings()
+  const monitorUserSession = !authSettings.scope?.includes('offline_access')
+  const urlRef = useRef<string>()
+  urlRef.current = `${window.location.pathname}${window.location.search}`
```

Comment:
Since we're not depending on `useLocation` anymore, we can get rid of `urlRef` and always get the latest location from `window.location`.